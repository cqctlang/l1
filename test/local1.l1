// this does not test language semantics;
// rather, it tests the property of the implementation
// that storage for locals is reused across disjoint blocks

define foo(){
	@local x;
	{
		@local x;
		x = 5;
	}
	{
		@local x;
		printf("%d\n", x);
	}	
	{
		@local x;
		printf("%d\n", x);
	}	
	{
		@local x;
		printf("%d\n", x);
	}	
	{
		@local x;
		printf("%d\n", x);
	}	
	{
		@local x;
		printf("%d\n", x);
	}	
	{
		@local x;
		printf("%d\n", x);
	}	
}

foo();
