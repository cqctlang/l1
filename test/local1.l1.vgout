entry: var params, 4 locals, 0 temps
L15: 0 params, 0 locals, 0 temps
L27: 1 params, 0 locals, 0 temps
0
1
2
3
4
5
6
7
8
9
