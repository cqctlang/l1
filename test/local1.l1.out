entry: var params, 4 locals, 0 temps
L12: 0 params, 0 locals, 0 temps
L20: 1 params, 0 locals, 0 temps
0
1
2
3
4
5
6
7
8
9
